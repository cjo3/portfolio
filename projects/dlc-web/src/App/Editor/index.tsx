import { Grid, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import clsx from 'clsx'
import React, { useState } from 'react'
import { FileUpload } from '../../interfaces'
import { TopNav } from '../TopNav'
import { FileLoader } from './FileLoader'
import { EditorSettings } from './EditorSettings'

const useStyles = makeStyles(theme => ({
  section: {
    padding: theme.custom.setSpace('sm')
  },
  fileLoaderBg: {
    background: `linear-gradient(top, ${theme.palette.primary[100]}, white)`
  },
  editorSettingsBg: {
    background: `linear-gradient(bottom, ${theme.palette.grey[900]}, ${theme.palette.grey[800]})`,
    boxShadow: '0.5rem 0.5rem 2rem rgba(0, 0, 0, 0.1)'
  }
}))

export const Editor = (): JSX.Element => {
  const classes = useStyles()

  const [loadedFile, setLoadedFile] = useState<FileUpload | null>(null)

  const unloadFile = (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ): void => {
    setLoadedFile(null)
  }

  return (
    <Grid container>
      <TopNav />
      <Grid component="section" container className={classes.section}>
        <Grid item xs={12}>
          <Typography variant="h1">Editor</Typography>
          <Typography variant="body1">
            Labore labore ut est sit sanctus stet et. Eos lorem ipsum consetetur
            magna est voluptua. Lorem justo dolor eirmod est est aliquyam. Dolor
            sit kasd nonumy lorem tempor dolor no duo et, invidunt gubergren
            rebum at rebum, et nonumy lorem lorem diam ipsum, sadipscing diam
            voluptua accusam sit sit, dolor.
          </Typography>
        </Grid>
      </Grid>
      {!loadedFile && (
        <Grid
          component="section"
          container
          className={clsx(classes.section, classes.fileLoaderBg)}>
          <FileLoader setLoadedFile={setLoadedFile} />
        </Grid>
      )}
      {loadedFile && (
        <Grid
          component="section"
          container
          className={clsx(classes.section, classes.editorSettingsBg)}>
          <EditorSettings />
        </Grid>
      )}
    </Grid>
  )
}
