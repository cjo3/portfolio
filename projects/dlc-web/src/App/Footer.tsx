import { PrivacyPolicy } from '@cjo3/shared/react/components/PrivacyPolicy'
import { TermsAndConditions } from '@cjo3/shared/react/components/TermsAndConditions'
import { ImageHandler } from '@cjo3/shared/react/components/ImageHandler'
import { switchLinkRoutePath } from '@cjo3/shared/react/helpers'
import { Button, Grid, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import GavelIcon from '@material-ui/icons/Gavel'
import VpnLockIcon from '@material-ui/icons/VpnLock'
import WebIcon from '@material-ui/icons/Web'
import clsx from 'clsx'
import React, { useState } from 'react'
import { ProfilePic } from '../../assets/'

const useStyles = makeStyles(theme => ({
  Footer_container: {
    padding: theme.custom.setSpace('sm'),
    background: theme.custom.setLinearGradient(
      180,
      theme.palette.grey[900],
      theme.palette.grey[800]
    )
  },
  Footer_linkButton: {
    ...theme.custom.setFlex('row', 'flext-start', 'center'),
    'marginTop': theme.custom.setSpace(),
    '&:first-child': {
      marginTop: 0
    }
  },
  Footer_linkButtonIcon: {
    fontSize: theme.typography.fontSize * 1.5,
    marginRight: theme.custom.setSpace() / 2,
    position: 'relative',
    top: -2
  },
  Footer_badgeTitleImageContainer: {
    ...theme.custom.setFlex('row', 'flex-end'),
    [theme.breakpoints.down('xs')]: {
      ...theme.custom.setFlex('row', 'flex-start'),
      margin: `${theme.custom.setSpace('sm')}px 0 0 0`
    },
    '&:hover': {
      cursor: 'pointer'
    }
  },
  Footer_badgeProfilePic: {
    'order': 2,
    'borderRadius': theme.custom.setSpace() / 4,
    [theme.breakpoints.down('xs')]: {
      order: 1
    },
    'transition': 'all 250ms ease-out',
    '&:hover': {
      boxShadow: theme.custom.boxShadow
    }
  },
  Footer_badgeTitle: {
    'textAlign': 'right',
    'order': 1,
    'marginRight': theme.custom.setSpace(),
    [theme.breakpoints.down('xs')]: {
      order: 2,
      textAlign: 'left',
      marginLeft: theme.custom.setSpace(),
      marginRight: 0
    },
    'transition': 'all 250ms ease-out',
    '&:hover': {
      color: theme.palette.primary[400]
    }
  },
  Footer_badgeSubtitle: {
    color: theme.palette.grey[400],
    textAlign: 'right',
    [theme.breakpoints.down('xs')]: {
      textAlign: 'left'
    }
  },
  Footer_copyright: {
    marginTop: theme.custom.setSpace('sm'),
    textAlign: 'right',
    [theme.breakpoints.down('xs')]: {
      textAlign: 'left'
    }
  }
}))

interface Props {
  style: string
}

export const Footer: React.FC<Props> = ({ style }): JSX.Element => {
  const classes = useStyles()

  const [TAndCOpen, setTAndCOpen] = useState<boolean>(false)

  const [PPOpen, setPPOpen] = useState<boolean>(false)

  const openTAndCHandler = (): void => setTAndCOpen(true)

  const closeTAndCHandler = (): void => setTAndCOpen(false)

  const openPPHandler = (): void => setPPOpen(true)

  const closePPHandler = (): void => setPPOpen(false)

  const badgeClickHandler = (): void => {
    if (window)
      window.open(process.env.SITE_URL, '_blank') ||
        window.location.replace(process.env.SITE_URL)
  }

  const homeLinkHandler = (): void => {
    if (window)
      window.location.replace(
        switchLinkRoutePath('/', process.env.APP_ROOT_PATH)
      )
  }

  return (
    <Grid
      container
      component="footer"
      className={clsx(style, classes.Footer_container)}>
      {TAndCOpen && (
        <TermsAndConditions
          open={TAndCOpen}
          closeHandler={closeTAndCHandler}
          siteName={process.env.SITE_NAME}
          siteUrl={process.env.SITE_URL}
          siteContactEmail={process.env.SITE_CONTACT_EMAIL}
        />
      )}
      {PPOpen && (
        <PrivacyPolicy
          open={PPOpen}
          closeHandler={closePPHandler}
          siteName={process.env.SITE_NAME}
          siteUrl={process.env.SITE_URL}
          siteContactEmail={process.env.SITE_CONTACT_EMAIL}
        />
      )}
      <Grid item xs={12} sm={6}>
        <Grid
          container
          justify="flex-start"
          alignItems="flex-start"
          direction="column">
          <Button
            type="button"
            color="primary"
            onClick={homeLinkHandler}
            className={classes.Footer_linkButton}>
            <WebIcon className={classes.Footer_linkButtonIcon} />
            Home
          </Button>
          <Button
            type="button"
            color="primary"
            className={classes.Footer_linkButton}
            onClick={openTAndCHandler}>
            <GavelIcon className={classes.Footer_linkButtonIcon} />
            Terms &amp; Conditions
          </Button>
          <Button
            type="button"
            color="primary"
            className={classes.Footer_linkButton}
            onClick={openPPHandler}>
            <VpnLockIcon className={classes.Footer_linkButtonIcon} />
            Privacy Policy
          </Button>
        </Grid>
      </Grid>
      <Grid item xs={12} sm={6}>
        <Grid
          container
          className={classes.Footer_badgeTitleImageContainer}
          onClick={badgeClickHandler}>
          <Typography
            variant="h5"
            color="primary"
            className={classes.Footer_badgeTitle}>
            Need Help with JavaScript
            <br />
            App Development?
          </Typography>
          <ImageHandler
            asset={ProfilePic}
            styleClass={classes.Footer_badgeProfilePic}
          />
        </Grid>
        <Typography variant="body1" className={classes.Footer_badgeSubtitle}>
          Available for Hire!
        </Typography>
        <Typography
          variant="body1"
          color="primary"
          className={classes.Footer_copyright}>
          &copy; 2020 {process.env.SITE_NAME}. All rights reserved.
        </Typography>
      </Grid>
    </Grid>
  )
}
