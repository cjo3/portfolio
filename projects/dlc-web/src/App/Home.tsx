import { Header } from '@cjo3/shared/react/components/Header'
import { Grid } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import React from 'react'
import { HomeHeader } from '../assets'

const useStyles = makeStyles(theme => ({
  Home_container: {
    height: '100%'
  }
}))

export const Home: React.FC = (): JSX.Element => {
  const classes = useStyles()
  return (
    <Grid container className={classes.Home_container}>
      <Header
        title={process.env.APP_NAME}
        subTitle="Quickly change detection limit values"
        bgColor="theme.palette.primary.main"
        bgUrl={HomeHeader.paths[0]}
        buttonHref="/editor"
        buttonLabel="Start"
      />
    </Grid>
  )
}
