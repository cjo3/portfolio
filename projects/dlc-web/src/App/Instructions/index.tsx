import React from 'react'
import { Button, Grid, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  Instructions_header: {
    padding: theme.custom.setSpace('sm')
  }
}))

interface Props {}

export const Instructions: React.FC<Props> = (): JSX.Element => {
  const classes = useStyles()
  return (
    <Grid container>
      <Grid item component="header" className={classes.Instructions_header}>
        <Typography variant="h1" align="center">
          Instructions
        </Typography>
        <Typography variant="body1" align="center">
          Labore labore ut est sit sanctus stet et. Eos lorem ipsum consetetur
          magna est voluptua. Lorem justo dolor eirmod est est aliquyam. Dolor
          sit kasd nonumy lorem tempor dolor no duo et, invidunt gubergren rebum
          at rebum, et nonumy lorem lorem diam ipsum, sadipscing diam voluptua
          accusam sit sit, dolor.
        </Typography>
      </Grid>
    </Grid>
  )
}
