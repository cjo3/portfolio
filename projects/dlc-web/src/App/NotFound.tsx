import { switchLinkRoutePath } from '@cjo3/shared/react/helpers'
import { Button, Grid, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import React from 'react'

const useStyles = makeStyles(theme => ({
  NotFound_contentContainer: {
    minHeight: '100%'
  },
  NotFound_header: {
    ...theme.custom.setFlex('column'),
    padding: theme.custom.setSpace('md'),
    backgroundImage:
      'url(https://image.freepik.com/free-photo/rendering-abstract-futuristic-background-with-glowing-neon-blue-orange-lights_181624-19807.jpg)',
    backgroundPosition: 'center',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
    color: theme.palette.grey[50]
  },
  NotFound_startButton: {
    marginTop: theme.custom.setSpace('sm')
  }
}))

export const NotFound: React.FC = (): JSX.Element => {
  const classes = useStyles()
  return (
    <Grid container className={classes.NotFound_contentContainer}>
      <Grid item xs={12} component="header" className={classes.NotFound_header}>
        <Typography variant="h1" align="center">
          Error
        </Typography>
        <Typography variant="body1" align="center">
          Sorry, something went wrong
        </Typography>
        <Button
          type="button"
          variant="contained"
          href={switchLinkRoutePath('/', process.env.APP_ROOT_PATH)}
          color="secondary"
          className={classes.NotFound_startButton}>
          Home
        </Button>
      </Grid>
    </Grid>
  )
}
