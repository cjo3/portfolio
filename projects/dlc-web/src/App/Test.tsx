import { processSheet } from '@cjo3/shared/logic/dlc'
import { sheetData as sheetDataMock } from '@cjo3/shared/react/mocks/dlc'
import { sortAddresses } from '@cjo3/shared/react/xlsx'
import { Grid, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import React from 'react'

const useStyles = makeStyles(theme => ({
  padding: {
    padding: theme.custom.setSpace('sm')
  }
}))

export const Test: React.FC = (): JSX.Element => {
  const classes = useStyles()

  const values = {
    rangeStart: 'B17',
    rangeEnd: 'AM47',
    ulTrigger: '<',
    ulTriggerZero: 'Rock',
    ulTransform: 'halve',
    olTrigger: '>',
    olTransform: 'halve'
  }

  const result = processSheet(sheetDataMock, values)

  console.log('%c result', 'color: yellow; font-size: large', result)

  return (
    <Grid container>
      <Grid item className={classes.padding}>
        <Typography variant="body1" align="center"></Typography>
      </Grid>
    </Grid>
  )
}
