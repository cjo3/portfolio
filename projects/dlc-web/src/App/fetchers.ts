import { post } from 'axios'
import { addAuthHeaderToOptions } from '@cjo3/shared/security/authToken'

const otherHeaders = {
  headers: {
    accept: 'application/json'
  }
}

export const makePreOrder = async () => {
  let apiUrl = 'http://localhost:2000'

  if (process.env.NODE_ENV === 'production')
    apiUrl = process.env.API_URL + '/detection-limit-corrector'

  const res = await post(
    `${apiUrl}/pre-order`,
    null,
    addAuthHeaderToOptions(otherHeaders)
  )
  return res.data
}
