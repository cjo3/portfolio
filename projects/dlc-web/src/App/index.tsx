import { CssBaseline } from '@material-ui/core'
import React from 'react'
import { Route, Switch } from 'react-router-dom'
import { Editor } from './Editor'
import { Home } from './Home'
import { NotFound } from './NotFound'
import { Footer } from './Footer'
import { Preview } from './Editor/Preview/'

export const App: React.FC = (): JSX.Element => (
  <CssBaseline>
    <Switch>
      <Route path="/" exact component={Home} />
      <Route path="/editor" exact component={Editor} />
      <Route path="/test" exact component={Preview} />
      <Route path="/*" component={NotFound} />
    </Switch>
    <Footer />
  </CssBaseline>
)
