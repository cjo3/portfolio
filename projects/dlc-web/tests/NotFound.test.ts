import { testRenderer } from './testRenderer'
import { NotFound } from '../src/App/NotFound'

describe('NotFound', () => {
  describe('render', () => {
    test('renders heading', () => {
      const { getByRole } = testRenderer('/error', NotFound)
      const h1 = getByRole('heading', { level: 1, name: 'Error' })
      expect(h1).toBeInTheDocument()
    })

    test('renders copy', () => {
      const { getByText } = testRenderer('/error', NotFound)
      const copy = getByText('sorry', {
        selector: 'p',
        exact: false
      })
      expect(copy).toBeInTheDocument()
    })

    test('renders link to home', () => {
      const { getByRole } = testRenderer('/error', NotFound)
      const homeLink = getByRole('link', { name: /home/i })
      expect(homeLink).toHaveAttribute('href', '/')
    })
  })

  describe('events', () => {
    test('link clicks to home', () => {
      const {
        getByRole,
        userEvent: { click }
      } = testRenderer('/error')
      const homeLink = getByRole('link', { name: /home/i })
      click(homeLink)
      const startLink = getByRole('link', { name: /start/i })
      expect(startLink).toBeInTheDocument()
    })
  })
})
