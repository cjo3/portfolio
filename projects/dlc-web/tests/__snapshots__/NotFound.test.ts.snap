// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound render matches current snapshop 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <section
    className="MuiGrid-root makeStyles-section-1 MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h1
        className="MuiTypography-root MuiTypography-h1"
      >
        Error
      </h1>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Sorry, an error occurred. Please try again later!
      </p>
    </div>
    <div
      className="MuiGrid-root makeStyles-homeLink-2 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
  </section>
</div>
`;
